src/wallet/wallet.h

```diff
@@ -751,6 +750,8 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
     MasterKeyMap mapMasterKeys;
     unsigned int nMasterKeyMaxID;
 
+    CHDMintWallet* zwallet;
```

Comment:
This will cause memory leak on some places due to it re-assign this without freeing the previos value. Better to to use `std::unique_ptr` instead.

Reply:
The value will not be reassigned within the same instance ever, but a `unique_ptr` is better anyway. Updated, please take a look